exporter/splunkhecexporter/integration_test.go

```diff
@@ -0,0 +1,398 @@
+}
+
+func prepareLogs() plog.Logs {
+	logs := plog.NewLogs()
+	logs.ResourceLogs().EnsureCapacity(1)
+	rl := logs.ResourceLogs().AppendEmpty()
+	rl.ScopeLogs().EnsureCapacity(1)
+
+	sl := rl.ScopeLogs().AppendEmpty()
+	sl.Scope().SetName("test")
+	sl.LogRecords().EnsureCapacity(1)
+
+	ts := pcommon.Timestamp(int64(0) * time.Millisecond.Nanoseconds())
+
+	logRecord := sl.LogRecords().AppendEmpty()
+	logRecord.Body().SetStr("test log")
+	logRecord.Attributes().PutStr(splunk.DefaultNameLabel, "test- label")
+	logRecord.Attributes().PutStr(conventions.AttributeHostName, "myhost")
+	logRecord.Attributes().PutStr("custom", "custom")
+	logRecord.SetTimestamp(ts)
+	return logs
+}
+
+func prepareLogsNonDefaultParams(index string, source string, sourcetype string, event string) plog.Logs {
+	logs := plog.NewLogs()
+	logs.ResourceLogs().EnsureCapacity(1)
+	rl := logs.ResourceLogs().AppendEmpty()
+	rl.ScopeLogs().EnsureCapacity(1)
+
+	sl := rl.ScopeLogs().AppendEmpty()
+	sl.Scope().SetName("test")
+	sl.LogRecords().EnsureCapacity(1)
+
+	ts := pcommon.Timestamp(int64(0) * time.Millisecond.Nanoseconds())
+
+	logRecord := sl.LogRecords().AppendEmpty()
+	logRecord.Body().SetStr(event)
+	logRecord.Attributes().PutStr(splunk.DefaultNameLabel, "label")
+	logRecord.Attributes().PutStr(splunk.DefaultSourceLabel, source)
+	logRecord.Attributes().PutStr(splunk.DefaultSourceTypeLabel, sourcetype)
+	logRecord.Attributes().PutStr(splunk.DefaultIndexLabel, index)
+	logRecord.Attributes().PutStr(conventions.AttributeHostName, "myhost")
+	logRecord.Attributes().PutStr("custom", "custom")
+	logRecord.SetTimestamp(ts)
+	return logs
+}
+
+func prepareMetricsData(metricName string) pmetric.Metrics {
+	metricData := pmetric.NewMetrics()
+	metric := metricData.ResourceMetrics().AppendEmpty().ScopeMetrics().AppendEmpty().Metrics().AppendEmpty()
+	g := metric.SetEmptyGauge()
+	g.DataPoints().AppendEmpty().SetDoubleValue(132.929)
+	metric.SetName(metricName)
+	return metricData
+}
+
+func prepareTracesData(index string, source string, sourcetype string) ptrace.Traces {
+	ts := pcommon.Timestamp(int64(0) * time.Millisecond.Nanoseconds())
+
+	traces := ptrace.NewTraces()
+	rs := traces.ResourceSpans().AppendEmpty()
+	rs.Resource().Attributes().PutStr("com.splunk.source", source)
+	rs.Resource().Attributes().PutStr("host.name", "myhost")
+	rs.Resource().Attributes().PutStr("com.splunk.sourcetype", sourcetype)
+	rs.Resource().Attributes().PutStr("com.splunk.index", index)
+	ils := rs.ScopeSpans().AppendEmpty()
+	initSpan("myspan", ts, ils.Spans().AppendEmpty())
+	return traces
+}
+
+func TestSplunkHecExporterEventsToSplunk(t *testing.T) {
+	logger.Println("Starting")
+	logger.Println("Test -> Splunk running at:", integrationtestutils.ViperGetConfigVariable("HOST"),
+		integrationtestutils.ViperGetConfigVariable("UI_PORT"),
+		integrationtestutils.ViperGetConfigVariable("HEC_PORT"),
+		integrationtestutils.ViperGetConfigVariable("MANAGEMENT_PORT"),
+	)
```

Comment:
This is going to be really noisy, can I suggest that you use `zaptest.NewLogger`  instead?

Reply:
I switched to`zaptest.NewLogger` as recommended